core/controllers/reader.py

```diff
@@ -81,6 +94,24 @@ def test_can_play(self, exploration_id, **kwargs):
 
 def _get_exploration_player_data(
         exploration_id, version, collection_id, can_edit):
+    """Gets exploration metadata which is returned as a dictionary.
+
+    Args:
+        exploration_id: str. The ID of the exploration to get metadata for.
+        version: int or None. If specified, the version of the exploration to
+            get metadata for.
+        collection_id: str. The id of the collection which the exploration
+            belongs to.
+        can_edit: bool. Whether the exploration is editable by the learner.
+
+    Returns:
+        dict. A dictionary with metadata pertaining to the exploration
+            such as GADGET_SPECS, can_edit, title, version etc.
+
+    Raises:
+        Exception: The collection is not in current available collections list
```

Comment:
To be clear, I meant listing both. These are two separate cases for which there are exceptions raised.